From c37936698341e1bbbec2b9b62dfc64ddc51f3dfd Mon Sep 17 00:00:00 2001
From: Steve Kowalik <steven@wedontsleep.org>
Date: Wed, 25 Jun 2025 16:02:31 +1000
Subject: [PATCH] Stop using external mock module

Now that support for Python 2 has been dropped, we can switch to the
built-in unittest.mock module and drop one more external test
dependency.

Change-Id: Ie6e3b2c4047d00509053769fd5b49a1e4464e951
---
 test-requirements.txt                    | 1 -
 tests/jobs/test_assert.py                | 2 +-
 tests/jobs/test_config.py                | 2 +-
 tests/jobs/test_copy.py                  | 2 +-
 tests/jobs/test_count.py                 | 2 +-
 tests/jobs/test_create.py                | 2 +-
 tests/jobs/test_debug.py                 | 2 +-
 tests/jobs/test_delete.py                | 2 +-
 tests/jobs/test_disable.py               | 2 +-
 tests/jobs/test_enable.py                | 2 +-
 tests/jobs/test_get.py                   | 2 +-
 tests/jobs/test_getall.py                | 2 +-
 tests/jobs/test_info.py                  | 2 +-
 tests/jobs/test_name.py                  | 2 +-
 tests/jobs/test_reconfig.py              | 2 +-
 tests/jobs/test_rename.py                | 2 +-
 tests/jobs/test_set_next_build_number.py | 2 +-
 tests/test_build.py                      | 2 +-
 tests/test_check_jenkinsfile_syntax.py   | 2 +-
 tests/test_credential.py                 | 2 +-
 tests/test_info.py                       | 2 +-
 tests/test_job_folder.py                 | 2 +-
 tests/test_node.py                       | 2 +-
 tests/test_plugins.py                    | 2 +-
 tests/test_queue.py                      | 2 +-
 tests/test_quiet_down.py                 | 2 +-
 tests/test_script.py                     | 2 +-
 tests/test_version.py                    | 2 +-
 tests/test_view.py                       | 2 +-
 tests/test_whoami.py                     | 2 +-
 30 files changed, 29 insertions(+), 30 deletions(-)

Index: python-jenkins-1.8.2/test-requirements.txt
===================================================================
--- python-jenkins-1.8.2.orig/test-requirements.txt
+++ python-jenkins-1.8.2/test-requirements.txt
@@ -1,7 +1,6 @@
 cmd2!=0.8.3,<0.9.0;python_version<'3.0'  # MIT
 cmd2!=0.8.3;python_version>='3.0'  # MIT
 coverage>=3.6
-mock
 unittest2
 python-subunit
 requests-mock>=1.11.0
Index: python-jenkins-1.8.2/tests/jobs/test_assert.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_assert.py
+++ python-jenkins-1.8.2/tests/jobs/test_assert.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_config.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_config.py
+++ python-jenkins-1.8.2/tests/jobs/test_config.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_copy.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_copy.py
+++ python-jenkins-1.8.2/tests/jobs/test_copy.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_count.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_count.py
+++ python-jenkins-1.8.2/tests/jobs/test_count.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_create.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_create.py
+++ python-jenkins-1.8.2/tests/jobs/test_create.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_debug.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_debug.py
+++ python-jenkins-1.8.2/tests/jobs/test_debug.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_delete.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_delete.py
+++ python-jenkins-1.8.2/tests/jobs/test_delete.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_disable.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_disable.py
+++ python-jenkins-1.8.2/tests/jobs/test_disable.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_enable.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_enable.py
+++ python-jenkins-1.8.2/tests/jobs/test_enable.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_get.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_get.py
+++ python-jenkins-1.8.2/tests/jobs/test_get.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.helper import build_response_mock
Index: python-jenkins-1.8.2/tests/jobs/test_getall.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_getall.py
+++ python-jenkins-1.8.2/tests/jobs/test_getall.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsGetJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_info.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_info.py
+++ python-jenkins-1.8.2/tests/jobs/test_info.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.helper import build_response_mock
Index: python-jenkins-1.8.2/tests/jobs/test_name.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_name.py
+++ python-jenkins-1.8.2/tests/jobs/test_name.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_reconfig.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_reconfig.py
+++ python-jenkins-1.8.2/tests/jobs/test_reconfig.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_rename.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_rename.py
+++ python-jenkins-1.8.2/tests/jobs/test_rename.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/jobs/test_set_next_build_number.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/jobs/test_set_next_build_number.py
+++ python-jenkins-1.8.2/tests/jobs/test_set_next_build_number.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.jobs.base import JenkinsJobsTestBase
Index: python-jenkins-1.8.2/tests/test_build.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_build.py
+++ python-jenkins-1.8.2/tests/test_build.py
@@ -1,7 +1,7 @@
 import json
 
 import collections
-from mock import patch, Mock
+from unittest.mock import patch, Mock
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_check_jenkinsfile_syntax.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_check_jenkinsfile_syntax.py
+++ python-jenkins-1.8.2/tests/test_check_jenkinsfile_syntax.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 
 from tests.base import JenkinsTestBase
 from tests.helper import build_response_mock
Index: python-jenkins-1.8.2/tests/test_credential.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_credential.py
+++ python-jenkins-1.8.2/tests/test_credential.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_info.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_info.py
+++ python-jenkins-1.8.2/tests/test_info.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_job_folder.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_job_folder.py
+++ python-jenkins-1.8.2/tests/test_job_folder.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_node.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_node.py
+++ python-jenkins-1.8.2/tests/test_node.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 import requests_mock
Index: python-jenkins-1.8.2/tests/test_plugins.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_plugins.py
+++ python-jenkins-1.8.2/tests/test_plugins.py
@@ -32,7 +32,7 @@
 
 
 import json
-from mock import patch
+from unittest.mock import patch
 from testscenarios.scenarios import multiply_scenarios
 
 import jenkins
Index: python-jenkins-1.8.2/tests/test_queue.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_queue.py
+++ python-jenkins-1.8.2/tests/test_queue.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_quiet_down.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_quiet_down.py
+++ python-jenkins-1.8.2/tests/test_quiet_down.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_script.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_script.py
+++ python-jenkins-1.8.2/tests/test_script.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_version.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_version.py
+++ python-jenkins-1.8.2/tests/test_version.py
@@ -1,4 +1,4 @@
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_view.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_view.py
+++ python-jenkins-1.8.2/tests/test_view.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
Index: python-jenkins-1.8.2/tests/test_whoami.py
===================================================================
--- python-jenkins-1.8.2.orig/tests/test_whoami.py
+++ python-jenkins-1.8.2/tests/test_whoami.py
@@ -1,5 +1,5 @@
 import json
-from mock import patch
+from unittest.mock import patch
 
 import jenkins
 from tests.base import JenkinsTestBase
